server/src/routes/replays.ts

```diff
@@ -153,4 +153,35 @@ router.post('/', (req: Request, res: Response, next: Function): any => {
     req.on('error', (err) => next(err));
 });
 
+/**
+ * DELETE /replays/:replayId
+ * Deletes a replay file
+ */
+router.delete('/:replayId', async (req, res) => {
+    // Fetch replay
+    const replay = await db.getReplayById(req.params.replayId);
+    const replayUser = await db.getUserById(replay.userRef);
+
+    // Check user
+    const { user } = req;
+    if (user === undefined || user.webId !== replayUser.webId) {
+        res.status(401).send('Not authorized to delete replay.');
+        return;
+    }
+
+    await db.deleteReplayById(replay._id);
+
+    try {
+        await artefacts.deleteReplay(replay);
+    } catch (err) {
+        // If deletion failed, add the replay back into the DB
+        await db.saveReplay(replay);
+
+        res.status(500).send('Failed to delete replay file.');
+        return;
+    }
+
+    res.status(200).end();
```

Comment:
Just a side note: I'm not sure if we have any consistency around `send` vs `end` - I like just always using `send` so you can never really go wrong. But it's very irrelevant tbf 😅

Reply:
Oh yeah, I will use `.send`. I just read that it internally uses the `.end()` call anyways, but sets headers accordingly to the data, so it's probably safer to use `res.send()` as opposed to `res.status(200).end()`